confdgnmi/testtool/gNMI_Interface/04__Subscribe.robot

```diff
@@ -5,90 +5,105 @@ Test Tags        subscribe
 Resource         Subscribe.resource
 
 Resource         gNMIClient.resource
-Suite Setup      Setup gNMI Client
-Suite Teardown   Close gNMI Client
+Test Setup       Setup gNMI Client
+Test Teardown    Teardown gNMI state
 
 
 *** Test Cases ***
 
 Basic subscription with "mode" parameter
     [Tags]    sanity
-    [Documentation]  Test that the device correctly responds to all three
-    ...    "Subscribe" request modes.  No further functionality (such as actually
+    [Documentation]    Test that the device correctly responds to all three
+    ...    "Subscribe" request modes.    No further functionality (such as actually
     ...    polling the device) is tested.
-    [Template]    Verify Subscribe
+    [Template]    Subscribe ${mode} to default path with encoding JSON_IETF
     STREAM
     ONCE
     POLL
 
-Subscribe to non-existent "prefix"
-
-Subscribe to non-existent "path"
-
-# message Subscription {
-#   Path path = 1;                    // The data tree path.
-#   SubscriptionMode mode = 2;        // Subscription mode to be used.
-#   uint64 sample_interval = 3;       // ns between samples in SAMPLE mode.
-#   bool suppress_redundant = 4;
-#   uint64 heartbeat_interval = 5;
-# }
-
 Subscribe ONCE with supported "encoding" values
-
-Two Subscriptions in single request - with same "path"
-
-Two Subscriptions in single request - with different "path"
+    [Tags]    sanity
+    [Documentation]    Verify that the device is able to respond correctly for
+    ...    all declared encodings.
+    Given device capabilities
+    and subscription paths    ${GET-PATH}
+    Then subscribe ONCE with supported encodings
```

Comment:
KW name does not "resolve" to anything for reader -> it's unclear what the test checks 
(i assume some check is "hidden" inside of this keyword, but it should be part of the test, or kw should imply the check with its name according to "good tests" guides of ROBOT)